les/handler.go

```diff
@@ -1249,14 +1289,18 @@ func (pc *peerConnection) RequestHeadersByNumber(origin uint64, amount int, skip
 
 func (d *downloaderPeerNotify) registerPeer(p *peer) {
 	pm := (*ProtocolManager)(d)
-	pc := &peerConnection{
-		manager: pm,
-		peer:    p,
+	if pm.ulc == nil || p.isTrusted {
```

Comment:
You should register untrusted peers with the downloader too. You can sync from them if they have a head that the trusted peers have already validated. Also, the downloader can download headers from multiple peers once it has obtained the header skeleton from the primary peer so we should definitely register every useful peer we have.